### PR TITLE
homework: initialize we_detached

### PR DESCRIPTION
**Pull request by sshedi**
Fixes: CID#1469711 CID#1469712

CID 1469711 (#1 of 1): Uninitialized scalar variable (UNINIT)
14. uninit_use: Using uninitialized value we_detached.

CID 1469712 (#1 of 1): Unused value (UNUSED_VALUE)
returned_value: Assigning value from sd_id128_from_string(word + 2, &boot_id) to r here,
but that stored value is overwritten before it can be used.

### CODE REVIEW

File: src/home/homework-luks.c [hunk 1]
```diff
@@ -1621,7 +1621,6 @@ int home_deactivate_luks(UserRecord *h, HomeSetup *setup) {
                         return log_error_errno(r, "Failed to initialize cryptsetup context for %s: %m", setup->dm_name);
                 if (r == 0) {
                         log_debug("LUKS device %s has already been detached.", setup->dm_name);
-                        we_detached = false;
                 }
```
Comment (yuwata) [inline src/home/homework-luks.c]: Please drop unnecessary `{}`.

File: src/home/homework-luks.c [hunk 2]
```diff
@@ -1633,10 +1632,9 @@ int home_deactivate_luks(UserRecord *h, HomeSetup *setup) {
                 r = sym_crypt_deactivate_by_name(setup->crypt_device, setup->dm_name, 0);
                 if (ERRNO_IS_DEVICE_ABSENT(r) || r == -EINVAL) {
                         log_debug_errno(r, "LUKS device %s is already detached.", setup->dm_node);
-                        we_detached = false;
-                } else if (r < 0)
+                } else if (r < 0) {
                         return log_info_errno(r, "LUKS device %s couldn't be deactivated: %m", setup->dm_node);
-                else {
+                } else {
```
Comment (yuwata) [inline src/home/homework-luks.c]: Also here. Please drop unnecessary `{}`.

File: src/home/homework-luks.c [hunk 3]
```diff
@@ -1603,7 +1603,7 @@ int home_activate_luks(
 }
 
 int home_deactivate_luks(UserRecord *h, HomeSetup *setup) {
-        bool we_detached;
+        bool we_detached = false;
```
Comment (yuwata) [inline src/home/homework-luks.c:1606]: The issue is CID#1469711 should be false positive, as when `acquire_open_luks_device()` returns `1`, `setup->crypt_device` is set.

But the change looks good to me, as it is simpler and clear for readers.

File: src/libsystemd/sd-journal/sd-journal.c [hunk 4]
```diff
@@ -987,6 +987,8 @@ _public_ int sd_journal_seek_cursor(sd_journal *j, const char *cursor) {
                 case 'b':
                         boot_id_set = true;
                         r = sd_id128_from_string(word + 2, &boot_id);
+                        if (r < 0)
+                                return r;
```
Comment (yuwata) [inline src/libsystemd/sd-journal/sd-journal.c:991]: Please split the change into another commit.

**Review by yuwata — CHANGES_REQUESTED** [review]
Mostly good.
- Please split changes.
- Please follow our coding style.

File: src/home/homework-luks.c (hunk 2 above)
Comment (sshedi) [reply src/home/homework-luks.c]: I though the first `if` is using `{}` even though it has one call, so though of keeping it consistent.

File: src/home/homework-luks.c (hunk 1 above)
Comment (yuwata) [reply src/home/homework-luks.c]: Not addressed. Please drop `{}`. See our coding style.

File: src/home/homework-luks.c [hunk 5]
```diff
@@ -1633,7 +1632,6 @@ int home_deactivate_luks(UserRecord *h, HomeSetup *setup) {
                 r = sym_crypt_deactivate_by_name(setup->crypt_device, setup->dm_name, 0);
                 if (ERRNO_IS_DEVICE_ABSENT(r) || r == -EINVAL) {
                         log_debug_errno(r, "LUKS device %s is already detached.", setup->dm_node);
-                        we_detached = false;
                 } else if (r < 0)
```
Comment (yuwata) [inline src/home/homework-luks.c]: Here too.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Review by yuwata — APPROVED** [review]
LGTM.
